CactbotOverlay/CactbotOverlay.cs

```diff
@@ -247,6 +253,79 @@ public CactbotOverlay(CactbotOverlayConfig config)
       fast_update_timer_semaphore_.Release();
     }
 
+    public void EnableWatchers(bool check) {
+      this.watcher_state = check;
+    foreach (var watcher in this.watchers)
+        watcher.EnableRaisingEvents = check;
+    }
+ 
+    private void SetupFileWatcher(string url) {
+      if (url == "")
+        return;
+
+      foreach(var watcher in this.watchers)
+        watcher.Dispose();
+
+      List<string> path_list = new List<string>();
+      var overlay_full_path = url;
+      string base_path = null;
+
+      if (!url.Contains("://quisquous.github.io/cactbot/")) {
+        // Path of overlay if overlay is not remote!
+        overlay_full_path = Regex.Replace(overlay_full_path, @"file:[\\\/]+", "");
+        var overlay_path = Path.GetDirectoryName(url);
+        overlay_path = Regex.Replace(overlay_path, @"file:[\\\/]+", "");
+        base_path = Regex.Match(overlay_path, @"(.*cactbot)").Groups[1].ToString();
+
+        path_list.Add(base_path);
+
+        // Path of HTML resources if overlay is not remote!
+        string html = File.ReadAllText(overlay_full_path);
+        var html_path = Regex.Match(html, @"(?:src|href)=""(.*)[\\\/].*\..*""");
+        foreach(var match in html_path.Groups) {
+          string match_path = match.ToString();
+          if (match_path.StartsWith(".."))
+            match_path = overlay_path + "/" + match_path;
+          if (!match_path.Contains(base_path) && Directory.Exists(match_path) && !path_list.Contains(match_path))
+            path_list.Add(match_path);
+        }
+      }
+
+      // User Dir
+      string user_path;
+      Dictionary<string, string> local_UserFiles = new Dictionary<string, string>();
+      GetUserConfigDirAndFiles(out user_path, out local_UserFiles);
+      user_path = Regex.Replace(user_path, @"file:[\\\/]+", "");
+      if(base_path == null || !user_path.Contains(base_path))
+        path_list.Add(user_path);
+
+      watchers = new FileSystemWatcher[path_list.Count];
+      for (int i = 0; i < path_list.Count; i++) {
+        if (!Directory.Exists(path_list[i])) {
+            LogError($"Directory {path_list[i]}, (index: {i}) does not exist!");
+            continue;
+        }
+        var watcher = new FileSystemWatcher() {
+          Path = path_list[i],
+          NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.FileName,
+          IncludeSubdirectories = true,
+        };
+
+        watcher.Created += (o, e) => { Navigate(url); };
+        watcher.Deleted += (o, e) => { Navigate(url); };
+        watcher.Renamed += (o, e) => { Navigate(url); };
+        watcher.Changed += (o, e) => { Navigate(url); };
```

Comment:
I don't know if it matters, but if I save a file, I definitely get two reloads instead of just one.

Reply:
For the same overlay? 
You may get reloads for every overlay that has the file watcher activated and is located within the same `base_path`.

Reply:
Hmm.  I have two watchers, one for cactbot/ and one for cactbot/user/.  If I add a LogDebug message into the Changed callback, and then change a file that's in cactbot/ui/, I see that watcher trigger twice.  Does that not happen for you?

Reply:
Depending on how your text editor saves files, it can trigger multiple FS events. For example, if you have `atomic_save` enabled in Sublime Text, it creates a new file each time you save and replaces the old file with the new one ([here's a more detailed explanation](https://stackoverflow.com/questions/20634684/what-is-sublime-text-doing-when-i-save-a-file)).
Basically, the only way around this is [debouncing](https://lodash.com/docs/4.17.15#debounce) the reloading.